runtime/sema/checker.go

```diff
@@ -360,6 +385,10 @@ func (checker *Checker) checkTopLevelDeclarationValidity(declarations []ast.Decl
 	}
 
 	for _, declaration := range declarations {
+		if _, ok := declaration.(*ast.PragmaDeclaration); ok {
+			continue
+		}
+
```

Comment:
Instead of special-casing it here, maybe add pragma to [validTopLevelDeclarationsInTransaction](https://github.com/onflow/cadence/blob/bd4d65b982a5a6cb3d75d33715b5b1db90e47da4/runtime/runtime.go#L167)?

Reply:
Good idea, I didn't realize this exists. I added it in [6f3c9cf](https://github.com/onflow/cadence/pull/2355/commits/6f3c9cf196ecc6bb0caa5b032a5b3f1af1e9ad5a), and then noticed how this top-level declaration validity checking is quite wasteful and allocates maps, so removed the pragma-special casing and optimized it in [6f7cd51](https://github.com/onflow/cadence/pull/2355/commits/6f7cd512e6dccffa3cea217d4a86ec13224f3ebf)